### ISSUE TITLE
Output file name clash

### ISSUE THREAD

**Issue by huard** [bird-house/finch#148]
## Description

In Finch, netCDF outputs from indicators are called `out.nc`. 

So if process 1 creates `out.nc`, and process 2 uses it as an input, the temp directory for process 2 will have an input `out.nc`, and then will try to write it's own output in `out.nc`. If the input `out.nc` is not closed, then xarray raises an error because we are trying to write over an open file. 

We could make sure all input files are closed, or ensure unique names for outputs. 

@MatProv thoughts ?

**Comment by MatProv** [comment]
Making sure that the file is closed before proceeding to further calculations in process 2 would be a bottleneck if this check is done at the file input level of process 2. If done at the end of process 1 (aka not exiting process unless files are completely closed) that would be great. Isn't it the desired actual behaviour, since there are no reason why process 1 would touch the file when process 2 is triggered?

It makes sense to ensure unique names for output files in my opinion. I see no caveat to this method, we only need process 2 to be aware of this filename.

**Comment by huard** [comment]
Ok, I've added the list of input files to `_make_unique_drs_filename` to avoid this. 

Next problem is that `try_opendap` hangs. It sends a HEAD request, which PyWPS probably does not understand. Even though there is a timeout, it completely hangs the server. 

So in the example below, if `url` is an output hosted on `localhost`, this hangs forever. 
```
content_description = requests.head(url, timeout=5).headers.get(
            "Content-Description"
        )
```

**Comment by huard** [comment]
Scrap that, it's not requests hanging, but `netCDF4.Dataset`.

**Comment by huard** [comment]
Fixed in #146

**Comment by huard** [comment]
Note that it's a dirty fix, we may be missing some functionality from the original implementation.
